### ISSUE TITLE
Enhancement  in sample_labels

### ISSUE THREAD

**Issue by pablomm** [GAA-UAM/scikit-fda#165]
The current ``sample_labels`` argument of ``plot`` to group samples by colors is very restrictive, only accepts labels of the form *0, 1, 2, ... n_classes -1*  without skip any number.

Can be fixed using [LabelEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html) internally.

**To Reproduce**
Code to reproduce the behavior:

```python
from skfda.datasets import make_sinusoidal_process
fd = make_sinusoidal_process(n_samples=5)

fd.plot(sample_labels=[0, 0, 2, 2, 2])  # Not valid
fd.plot(sample_labels=[-1, -1, 0, 1, 1]) # Not valid
```
**Result**
> **ValueError**: sample_labels must contain at least an occurence of numbers between 0 and number of distint sample labels.
